### PR TITLE
feat(rules): added rule for prefer-to-have-class

### PR DESCRIPTION
**Pull request by benmonro**
**What**: a new rule for prefer-to-have-class

**Why**: all the matchers!

**How**:

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

### CODE REVIEW

**Comment by benmonro** [comment]
@AntonNiklasson mind giving this a once over?  if you think of any additional use cases let me know.

File: docs/rules/prefer-to-have-class.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# prefer toHaveClass over checking element.class (prefer-to-have-class)
+
+This rule is an autofixable rule that reports usages of checking element.class in expect statements in preference of using the jest-dom
```
Comment (AntonNiklasson) [inline docs/rules/prefer-to-have-class.md:3]:
```suggestion
This rule is an autofixable rule that reports usages of checking element.className in expect statements in preference of using the jest-dom
```

File: docs/rules/prefer-to-have-class.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+expect(el.className).not.toContain("baz");
+expect(el).toHaveAttribute("class", "qux");
+```
+
+Examples of **correct** code for this rule:
+
+```js
+expect(el).toHaveStyle({ foo: "bar" });
+expect(el.class).toMatchSnapshot();
+expect(el.class).toEqual(foo);
```
Comment (AntonNiklasson) [inline docs/rules/prefer-to-have-class.md:25]: Perhaps we could also include the actual correct "base case" here. Something like `expect(el).toHaveClass('foo');`

**Review by AntonNiklasson — COMMENTED** [review]
Just a few minor things. Also, did you consider supporting something like `el.classList.contains('foo')` here?

**Pull request merged**